src/luagit2/branch/lua_branch.c

```diff
@@ -0,0 +1,187 @@
+#include "lua_branch.h"
+
+int lua_git_branch_create (lua_State *L) {
+	luagit2_reference *lua_ref;
+	const luagit2_repository *lua_repo = (luagit2_repository *)lua_touserdata(L, 1);
+	const char *branch_name = luaL_checkstring(L, 2);
+	const luagit2_commit *lua_commit = (luagit2_commit *)lua_touserdata(L, 3);
+	const int force = luaL_checkinteger(L, 4);
+
+	lua_ref = (luagit2_reference *)lua_newuserdata(L, sizeof(*lua_ref));
+	lua_ref->reference  = NULL;
+
+	luaL_newmetatable(L, "luagit2_reference");
+
+	lua_setmetatable(L, -2);
+
+	git_reference *local_ref;
+	check_error_long(git_branch_create(&local_ref, lua_repo->repo, branch_name, lua_commit->commit, force),
+	    "Error creating new branch for target commit", NULL);
+	lua_ref->reference  = local_ref;
+
+	git_reference_free(local_ref);
```

Comment:
Isn't this going to destroy the object instead of just de-assigning it from local_ref?